yt/frontends/boxlib/fields.py

```diff
@@ -492,9 +493,12 @@ def setup_fluid_fields(self):
                 )
 
                 # Most of the time our species will be of the form
-                # element name + atomic weight (e.g. C12), but
-                # sometimes we make up descriptive names (e.g. ash)
-                if any(char.isdigit() for char in field):
+                # element name + atomic weight (e.g. X(C12)), but
+                # sometimes we make up descriptive names (e.g. X(ash)).
+                # In niche cases, we might have a molecule with a number
+                # in the middle (e.g. X(H2O)). Check if the last character before
+                # the ) is a digit.
+                if field[-2].isdigit():
```

Comment:
I still find this too fragile, but digging for a way to make it more robust, I think I ended up solving the problem of parsing molecules. Do you mind if I push to your branch ? I could also open a separate PR.

Reply:
Sure, whatever you think is best

Reply:
ok I'll set myself a reminder to execute this by next week, thank you for your patience !